egs/wsj/s5/steps/libs/nnet3/xconfig/parser.py

```diff
@@ -67,11 +67,59 @@ def xconfig_line_to_object(config_line, prev_layers = None):
             "*** {0}".format(config_line))
         raise
 
+# This function reads existing model (*.mdl) and returns array of
+# XconfigInputLayer one per input-node or component-node with same 'name' used
+# in model and 'dim' equal to 'output-dim' for component-node and 'input-dim' for
+# input-node.
+def get_model_component_info(model_filename):
+    all_layers = []
+    try:
+        f = open(model_filename, 'r')
+    except Exception as e:
+        sys.exit("{0}: error reading model file '{1}'".format(sys.argv[0], model_filename, repr(e)))
+
+    # use nnet3-info to get component-nodes and input-nodes in the model.
+    out = common_lib.get_command_stdout("""nnet3-info {0} | grep '\-node' """
+                                        """ """.format(model_filename))
+
+    # out contains all {input,component}-nodes used in model_filename
+    layer_names = []
+    for line in out.split("\n"):
+        parts = line.split(" ")
+        input_dim = -1
+        for  field in parts:
+            key_value = field.split("=")
+            if len(key_value) == 2:
+                key = key_value[0]
+                value = key_value[1]
+                if key == "name":
```

Comment:
Give examples for different types of lines that will be parsed. input-node, component-node etc.